### PR TITLE
*: make max backoff shorter

### LINKED ISSUE #7821: qps recover after 3mins when all tso recover from failure lasts for 10mins
### What did you do?
1、run tpcc
2、inject all tso failure last for 10mins
image

### What did you expect to see?
qps can recover quickly when fault recover

### What did you see instead?
qps recover after 3mins when all tso recover from failure lasts for 10mins
image

### What version of PD are you using (`pd-server -V`)?
./pd-server -V
 Release Version: v7.1.1-serverless
Edition: Community
Git Commit Hash: 1b93dce40464910fb6c44deb85363a838f4c8b8e
Git Branch: release-7.1-keyspace
UTC Build Time:  2024-02-05 08:28:18
2024-02-17T13:15:00.516+0800

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #7821.

### What is changed and how does it work?

This PR changes the gRPC connection backoff config to reduce the recovery time.

```commit-message
```

### Check List

Tests

- Integration test
Screenshot 2024-02-29 at 11 04 11

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/grpcutil/grpcutil.go [hunk 1]
```diff
@@ -62,7 +63,15 @@ func GetClientConn(ctx context.Context, addr string, tlsCfg *tls.Config, do ...g
 	if err != nil {
 		return nil, errs.ErrURLParse.Wrap(err).GenWithStackByCause()
 	}
-	cc, err := grpc.DialContext(ctx, u.Host, append(do, opt)...)
+	backoffOpts := grpc.WithConnectParams(grpc.ConnectParams{
```
Comment (lhy1024) [inline client/grpcutil/grpcutil.go:66]: why use this config? Do we need to add some comments here?

File: pkg/utils/grpcutil/grpcutil.go [hunk 2]
```diff
@@ -147,7 +148,16 @@ func GetClientConn(ctx context.Context, addr string, tlsCfg *tls.Config, do ...g
 	if err != nil {
 		return nil, errs.ErrURLParse.Wrap(err).GenWithStackByCause()
 	}
-	cc, err := grpc.DialContext(ctx, u.Host, append(do, opt)...)
+	backoffOpts := grpc.WithConnectParams(grpc.ConnectParams{
+		Backoff: backoff.Config{
+			BaseDelay:  100 * time.Millisecond,
```
Comment (CabinfeverB) [inline pkg/utils/grpcutil/grpcutil.go:153]: I found the BaseDelay is 1s in https://github.com/tikv/pd/blob/e199866f59e22e3759a8e9459ef33d57f784890d/pkg/syncer/client.go#L117.
Do we need keep them same?

Comment (rleungx) [reply pkg/utils/grpcutil/grpcutil.go]: ok

**Review by CabinfeverB — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm. and need some comments as @lhy1024  says.

**Review by lhy1024 — APPROVED** [review]

File: client/grpcutil/grpcutil.go [hunk 3]
```diff
@@ -62,7 +63,18 @@ func GetClientConn(ctx context.Context, addr string, tlsCfg *tls.Config, do ...g
...
 	}
-	cc, err := grpc.DialContext(ctx, u.Host, append(do, opt)...)
+	// Here we use a shorter MaxDelay to make the connection recover faster.
+	// The default MaxDelay is 120s, which is too long for us.
+	backoffOpts := grpc.WithConnectParams(grpc.ConnectParams{
+		Backoff: backoff.Config{
+			BaseDelay:  time.Second,
+			Multiplier: 1.6,
+			Jitter:     0.2,
+			MaxDelay:   3 * time.Second,
```
Comment (niubell) [inline client/grpcutil/grpcutil.go:73]: How about making MaxDelay configurable through some WithXX function to meet different callers‘s requests?

**Pull request merged**